atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolImpl.java

```diff
@@ -346,6 +346,23 @@ void setServersInPoolTo(ImmutableMap<CassandraServer, CassandraServerOrigin> des
                 .removeKeys(currentServers::contains)
                 .toImmutableMap();
         SetView<CassandraServer> absentServers = Sets.difference(currentServers, desiredServers.keySet());
+        Map<CassandraServer, CassandraClientPoolingContainer> currentPoolWithoutAbsentServers = EntryStream.of(
+                        getCurrentPools())
+                .removeKeys(absentServers::contains)
+                .toMap();
+
+        Set<CassandraServer> validatedServersToAdd =
+                validateNewHostsTopologiesAndMaybeAddToPool(currentPoolWithoutAbsentServers, serversToAdd);
+
+        Preconditions.checkState(
+                !(currentPoolWithoutAbsentServers.isEmpty() && validatedServersToAdd.isEmpty()),
+                "No servers were successfully added to the pool. This means we could not come to a consensus on"
+                        + " cluster topology, or no servers were provided. We will fallback to using our previous"
+                        + " Cassandra servers, if any exist. This state should be transient (<5 minutes), and if it is"
+                        + " not, indicates that the user may have accidentally configured AltasDB to use two separate"
+                        + " Cassandra clusters (i.e., user-led split brain).",
+                SafeArg.of("serversToAdd", CassandraLogHelper.collectionOfHosts(serversToAdd.keySet())),
```

Comment:
Could we increment a metric here? If that's not straightforward we don't need to do it in this PR, but this seems like something we could/would want to monitor. For example, if that metric was >0 for >5mins, something would be off.